overlord/snapstate/snapstate_test.go

```diff
@@ -5288,6 +5288,78 @@ func (s *snapmgrTestSuite) TestSnapStateLocalRevision(c *C) {
 	c.Assert(snapst.LocalRevision().Unset(), Equals, true)
 }
 
+func (s *snapmgrTestSuite) TestSnapStateLocalComponentRevision(c *C) {
+	si7 := snap.SideInfo{
+		RealName: "some-snap",
+		Revision: snap.R(-7),
+	}
+	csi1 := snap.NewComponentSideInfo(naming.NewComponentRef("some-snap", "mycomp"), snap.R(-3))
```

Comment:
Maybe put a non-local revision of the components in here, just to make sure that we ignore those?

Reply:
Good point, added now